### PR TITLE
💫 Added withNamedArgs method to the Chai emit matcher

### PR DESCRIPTION
**Pull request by naddison36**
Improved error messages for existing emit matchers

This is part of feature request https://github.com/EthWorks/Waffle/issues/437 to check against a subset of args emitted in events

### CODE REVIEW

File: waffle-chai/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@ethereum-waffle/chai",
   "description": "A sweet set of chai matchers for your blockchain testing needs.",
-  "version": "3.3.0",
+  "version": "3.3.1",
```
Comment (marekkirejczyk) [inline waffle-chai/package.json:4]: No need to update version - it will be updated by github bot (changesets).

**Comment by xiaolin** [comment]
Any plans to add something thing this? I am running into the same issue with a need to have an optional/unpredictable arg.

**Comment by marekkirejczyk** [comment]
@vanruch

**Comment by KholdStare** [comment]
I think this is an incredibly useful feature for any non-trivial contracts. Can someone take a look again from the waffle project and give directions? Any objections? Anything that can be fixed?

**Review by yivlad — APPROVED** [review]

**Pull request merged**